### PR TITLE
CORTX-30931: Implement REST API helper functionality and HTTP error code framework

### PR DESCRIPTION
**Pull request by akash2144**
Signed-off-by: Akash Dudhane <akash.a.dudhane@seagate.com>

# Problem Statement
- Implement REST API helper functionality and HTTP error code framework

# Design
-  https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/981434369/HA+REST+APIs
# Coding
-  [x] Coding conventions are followed and code is consistent

# Testing 
- [x] Unit and System Tests are added
- https://jts.seagate.com/secure/attachment/515419/test_result.txt
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM

# Review Checklist 
- [x] PR is self reviewed
- [x] JIRA number/GitHub Issue added to PR
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained
- [ ] Is there a change in filename/package/module or signature? [Y/N]: N
- [ ] If yes for above point, is a notification sent to all other cortx components? [Y/N] NA
- [ ] Side effects on other features (deployment/upgrade)? [Y/N] N
- [ ] Dependencies on other component(s)? [Y/N] N
-     If yes for above point, post link to the corresponding PR.

# Review Checklist 
- [ ] Is perfline test run and the report with and without the changes updated in the PR? [Y/N]: 

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by akash2144** [comment]
PR contains some changes from PR #696. Once that PR is merged will rebase this PR.

File: ha/fault_tolerance/rest_api.py [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+            except Exception as e:
+                Log.warn(f"Exception: {e}")
+                CcRestApi.error_response(Exception("Invalid endpoint."))
+
+            resp = await handler(request)
+
+            if isinstance(resp, web.StreamResponse):
+                return resp
+
+            status = 200
```
Comment (mariyappanp) [inline ha/fault_tolerance/rest_api.py:171]: successful status code can be in const.py

**Review by mariyappanp — COMMENTED** [review]
Do we have separate ticket for covering unittests? @akash2144

**Review by gauri-bhosale — COMMENTED** [review]
please resolve codecy issue

**Comment by akash2144** [comment]
> Do we have separate ticket for covering unittests? @akash2144

@mariyappanp Currently no API is available for testing, Once API is available will write test case there. [CORTX-30029](https://jts.seagate.com/browse/CORTX-30029)

**Comment by mariyappanp** [comment]
It would be better to maintain rest api framework if we separate it from ha service like ha/framework/restapi.
What do you think, @akash2144  @shaileshsvc @ArchanaLimaye

**Comment by hessio** [comment]
Is this PR still relevant? If not, can we close it?

**Comment by shailesh-vaidya** [comment]
Closing as an obsolete

**Pull request closed without merging**